docs/source/user_guide/gcm_based_inference/answering_causal_questions/quantify_intrinsic_causal_influence.rst

```diff
@@ -6,51 +6,64 @@ By quantifying intrinsic causal influence, we answer the question:
+which measures the influence of a node that has not been inherited from its parents, that is, the influence of the noise
+of a node. The reason for this is that, while ``Train C`` has to wait for ``Train B``, ``Train B`` mostly inherits the delay from
+``Train A``. Thus, ``Train A`` should be identified as the node that contributes the most to the delay of ``Train C``.
+
+See the :ref:`Understanding the method <understand-method>` section for another example and more details.
 
 How to use it
 ^^^^^^^^^^^^^^
 
-To see how the method works, let us generate some data.
+To see how the method works, let us generate some data following the example above:
 
 >>> import numpy as np, pandas as pd, networkx as nx
 >>> from dowhy import gcm
->>> from dowhy.gcm.uncertainty import estimate_variance
->>> np.random.seed(10)  # to reproduce these results
 
->>> X = np.random.normal(loc=0, scale=1, size=1000)
->>> Y = 2*X + np.random.normal(loc=0, scale=1, size=1000)
->>> Z = 3*Y + np.random.normal(loc=0, scale=1, size=1000)
+>>> X = abs(np.random.normal(loc=0, scale=5, size=1000))
+>>> Y = X + abs(np.random.normal(loc=0, scale=1, size=1000))
+>>> Z = Y + abs(np.random.normal(loc=0, scale=1, size=1000))
 >>> data = pd.DataFrame(data=dict(X=X, Y=Y, Z=Z))
 
-Next, we will model cause-effect relationships as a structural causal model and fit it to the data.
+Note the larger standard deviation of the 'noise' in :math:`X`.
+
+Next, we will model cause-effect relationships as a structural causal model and fit it to the data. Here, we are using
+the auto module to automatically assign causal mechanisms:
 
 >>> causal_model = gcm.StructuralCausalModel(nx.DiGraph([('X', 'Y'), ('Y', 'Z')])) # X -> Y -> Z
->>> causal_model.set_causal_mechanism('X', gcm.EmpiricalDistribution())
->>> causal_model.set_causal_mechanism('Y', gcm.AdditiveNoiseModel(gcm.ml.create_linear_regressor()))
->>> causal_model.set_causal_mechanism('Z', gcm.AdditiveNoiseModel(gcm.ml.create_linear_regressor()))
+>>> gcm.auto.assign_causal_mechanisms(causal_model, data)
 >>> gcm.fit(causal_model, data)
 
-..
-    Todo: Use auto module for automatic assignment!
-
 Finally, we can ask for the intrinsic causal influences of ancestors to a node of interest (e.g., :math:`Z`).
 
->>> contributions = gcm.intrinsic_causal_influence(causal_model, 'Z',
->>>                                               gcm.ml.create_linear_regressor(),
->>>                                               lambda x, _: estimate_variance(x))
+>>> contributions = gcm.intrinsic_causal_influence(causal_model, 'Z')
 >>> contributions
-    {'X': 33.34300732332951, 'Y': 9.599478688607254, 'Z': 0.9750701113403872}
+    {'X': 8.736841722582117, 'Y': 0.4491606897202768, 'Z': 0.35377942123477574}
+
+Note that, although we use a simple linear relationship here, the method can also handle arbitrary non-linear relationships.
 
 **Interpreting the results:** We estimated the intrinsic influence of ancestors of
```

Comment:
Nit [for consistency reason]: intrinsic causal influence

Reply:
Fixed